packages/mobile/src/recipients/reducer.ts

```diff
@@ -12,44 +12,55 @@ export interface State {
   // valoraRecipientCache contains accounts that the user has sent/recieved transactions from,
   // and includes CIP8 profile data if available
   valoraRecipientCache: AddressToRecipient
+  rewardsSenders: string[]
 }
 
-const initialState = {
+const initialState: State = {
   phoneRecipientCache: {},
   valoraRecipientCache: {},
+  rewardsSenders: [],
 }
 
-export const recipientsReducer = (
-  state: State | undefined = initialState,
-  action: ActionTypes | RehydrateAction
-) => {
-  switch (action.type) {
-    case REHYDRATE: {
+const rehydrate = createAction<any>(REHYDRATE)
+export const setPhoneRecipientCache = createAction<NumberToRecipient>(
+  'RECIPIENTS/SET_PHONE_RECIPIENT_CACHE'
+)
+export const updateValoraRecipientCache = createAction<AddressToRecipient>(
+  'RECIPIENTS/SET_VALORA_RECIPIENT_CACHE'
+)
+export const rewardsSendersFetched = createAction<string[]>('RECIPIENTS/REWARDS_SENDERS_FETCHED')
+
+export const recipientsReducer = createReducer(initialState, (builder) => {
+  builder
+    .addCase(rehydrate, (state, action) => {
+      // hack to allow rehydrate actions here
+      const hydrated = getRehydratePayload((action as unknown) as RehydrateAction, 'recipients')
       return {
+        ...initialState,
```

Comment:
I think `initialState` is not even needed, since `state` should be the `initialState` already.